File: src/TrafficLightSystem/TrafficLight.java
/**
 * 
 */
package trafficLightSystem;


/**
 * 
 */
public class TrafficLight {

	// vars
	private static int nextSystemID = 804;
	private int trafficLightID;
	private String state;
	private boolean status;
	
	// default constructor
	public TrafficLight() {
		//this.state = state;
		this.status = true;
		this.trafficLightID = ++nextSystemID;    // auto increment id
	};
	
	
	public TrafficLight(String state) {

		this.state = state;
		this.status = true;
		this.trafficLightID = ++nextSystemID;    // auto increment id
	}



	// setters
	
	public void setState(String state) {
		this.state = state;
	}
	
	public void setStatus(boolean status) {
		this.status = status;
	}
	

	// getters
	public int getTrafficLightID() {
		return trafficLightID;
	}


	public String getState() {
		return state;
	}


	public boolean isStatus() {
		return status;
	}

}

File: src/services/trafficLightSystem/TrafficLight.js
/**
 * 
 */
class TrafficLight {
    // vars
    static nextSystemID = 804;

    // constructor
    constructor(trafficLightSystemID, state = null) {
        this.trafficLightID = ++TrafficLight.nextSystemID;
        this.trafficLightSystemID = trafficLightSystemID;
        this.state = state;
        this.position = "";
        this.status = true;
    }

    // setters

    /**
     * Set TL state
     */
    setState(state) {
        this.state = state;
    }

    /**
     * Set TL status
     */
    setStatus(status) {
        this.status = status;
    }

    /**
     * Set TL position
     */
    setPosition(position) {
        this.position = position;
    }

    // getters

    /**
     * Get Traffic Light Id the TL is associated to
     */
    getTrafficLightID() {
        return this.trafficLightID;
    }

    /**
     * Get Traffic Light System Id the TL is associated to
     */
    getTrafficLightSystemID() {
        return this.trafficLightID;
    }

    /**
     * Get TL state
     */
    getState() {
        return this.state;
    }

    /**
     * Get TL position
     */
    getPosition() {
        return this.position;
    }

    /**
     * Get TL status
     */
    isStatus() {
        return this.status;
    }
}

// Export the TrafficLight class
module.exports = TrafficLight;
